fennel/client/client.py

```diff
@@ -444,22 +386,39 @@ def extract_historical_features(
         self,
         input_feature_list: List[Union[Feature, Featureset]],
         output_feature_list: List[Union[Feature, Featureset]],
-        input_dataframe: pd.DataFrame,
-        timestamps: pd.Series,
-    ) -> Union[pd.DataFrame, pd.Series]:
+        timestamp_column: str,
+        format: str = "pandas",
+        input: Dict[str, Any] = {},
+    ) -> Dict[str, Any]:
         """
         Extract point in time correct features from a dataframe, where the
         timestamps are provided by the timestamps parameter.
 
         Parameters:
         input_feature_list (List[Union[Feature, Featureset]]): List of features or featuresets to use as input.
         output_feature_list (List[Union[Feature, Featureset]]): List of features or featuresets to compute.
-        input_dataframe (pd.DataFrame): Dataframe containing the input features.
-        timestamps (pd.Series): Timestamps for each row in the input dataframe.
+        timestamp_column (str): The name of the column containing the timestamps.
+        format (str): The format of the input data. Can be either "pandas",
+        "csv", "json" or "parquet". Default is "pandas".
```

Comment:
nit: consider adding indentation in a multi-line doc string to avoid make the parameter list more readable